### PR TITLE
pkg/storage: enable EFOS, excise on snapshot, ingest splitting

### LINKED ISSUE #115432: storage: enable excise-on-snapshot and EFOS by default
We have two cluster settings:
- `storage.experimental.eventually_file_only_snapshots.enabled`,
- `kv.snapshot_receiver.excise.enabled`,
- `storage.ingest_split.enabled`.

All are disabled by default in 23.2.

Change the defaults to `true` on `master`.

Jira issue: CRDB-34032

### PR DESCRIPTION
**Pull request by nicktrav**
The following three cluster settings were introduced in the 23.2 cycle, and are disabled by default in that release:

- `storage.experimental.eventually_file_only_snapshots.enabled`,
- `kv.snapshot_receiver.excise.enabled`,
- `storage.ingest_split.enabled`.

Enable all three by default, in preparation for 24.1.

Fixes #115432.

### CODE REVIEW

**Comment by nicktrav** [comment]
Going to wait until cockroachdb/pebble#3147 makes its way up into Cockroach before landing this.

**Review by itsbilal — APPROVED** [review]
LGTM once the underlying bugfix lands

File: pkg/storage/pebble.go
Comment (itsbilal) [inline pkg/storage/pebble.go:147]: nit: it currently sounds like _disabling_ ingest-time splitting is the experimental option. We could probably reword this as "set to false to disable experimental ingest-time splitting that lowers write-amp", or drop the experimental altogether seeing as if it's defaulting to true, we should be able to stand behind it by the time 24.1 comes out.

Comment (nicktrav) [reply pkg/storage/pebble.go]: Done.

**Review by itsbilal — APPROVED** [review]
LGTM

**Comment by itsbilal** [comment]
#116892 will unblock the tests in this PR.

**Comment by itsbilal** [comment]
Closing as #117116 has landed.

**Pull request closed without merging**
